seatunnel-translation/seatunnel-translation-spark/seatunnel-translation-spark-common/src/main/java/org/apache/seatunnel/translation/spark/common/serialization/InternalRowConverter.java

```diff
@@ -164,11 +168,11 @@ private static Object reconvert(Object field, SeaTunnelDataType<?> dataType) {
             case ROW:
                 return reconvert((InternalRow) field, (SeaTunnelRowType) dataType);
             case DATE:
-                return ((Date) field).toLocalDate();
+                return LocalDate.ofEpochDay((int) field);
             case TIME:
-                return ((Timestamp) field).toLocalDateTime().toLocalTime();
+                return LocalDateTime.ofInstant(Instant.ofEpochMilli((long) field), ZONE_UTC).toLocalTime();
             case TIMESTAMP:
-                return ((Timestamp) field).toLocalDateTime();
+                return LocalDateTime.ofInstant(Instant.ofEpochMilli((long) field), ZONE_UTC);
```

Comment:
The same as above.

Reply:
Can the accuracy of TimesStamp accurately be precise?